beacon-chain/sync/pending_blocks_queue.go

```diff
@@ -279,28 +300,76 @@ func (s *Service) deleteBlockFromPendingQueue(slot uint64, b *ethpb.SignedBeacon
 		newBlks = append(newBlks, blk)
 	}
 	if len(newBlks) == 0 {
-		delete(s.slotToPendingBlocks, slot)
-		return
+		s.slotToPendingBlocks.Delete(slotToCacheKey(slot))
+		return nil
+	}
+
+	// Decrease exp itme in proportion to how many blocks are still in the cache for slot key.
+	d := time.Duration(gcache.DefaultExpiration.Seconds()/float64(len(newBlks))) * time.Second
```

Comment:
Default expiration is actually 0, so it needs to be changed to our actual expiration period